include/gatekeeper_gk.h

```diff
@@ -122,6 +140,13 @@ struct gk_fib {
 			 */
 			struct ether_cache eth_cache;
 		} grantor;
+
+		/*
+		 * When the action is GK_FWD_BACK_NET or GK_FWD_FRONT_NET,
+		 * we need the next fib entry for the gateway of the
+		 * destination.
+		 */
+		struct gk_fib *next_fib;
```

Comment:
Do you mean that we will maintain a hash table for the neighbor entries, which should maintain a simple hash table. Each bucket will maintain the Ethernet information for that a neighbor? Therefore, both the grantor entries and gateway entries can simply have a reference to the entry in the neighbor entries instead of maintaining by themselves. For IPv4, the table size is reasonable. How about IPv6? Does the simple hash function still work?